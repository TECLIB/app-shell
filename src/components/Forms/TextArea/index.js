import React from 'react'
import PropTypes from 'prop-types'

/**
 * Component to create a textArea input
 * @function TextArea
 * @param {object} props
 * @return {component}
 */
const TextArea = props => (
  <div className="froms__col">
    <p>
      {props.label}
    </p>
    <textarea
      rows={props.rows}
      className="win-textarea"
      name={props.name}
      value={(props.value || undefined)}
      placeholder={props.placeholder}
      onChange={event => props.function(props.name, event.target.value)}
      disabled={props.disabled}
      style={props.style}
      required={props.required}
    />
  </div>
)

TextArea.defaultProps = {
  label: '',
  required: false,
  rows: 6,
  placeholder: null,
  function: () => {},
  style: {},
  disabled: false,
  value: undefined,
}

TextArea.propTypes = {
  label: PropTypes.string,
  name: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  value: PropTypes.string,
  placeholder: PropTypes.string,
  function: PropTypes.func,
  disabled: PropTypes.bool,
  style: PropTypes.object,
  required: PropTypes.bool,
  rows: PropTypes.number,
}

export default TextArea
