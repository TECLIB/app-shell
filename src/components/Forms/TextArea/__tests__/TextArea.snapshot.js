import React from 'react'
import renderer from 'react-test-renderer'
import TextArea from '../index'

describe('TextArea', () => {
  test('renders TextArea', () => {
    const component = renderer.create(
      <TextArea
        name="test"
        label="test"
        value="test"
        placeholder="test"
        rows={8}
        function={() => {}}
        style={{ height: '40px' }}
        required
      />,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
