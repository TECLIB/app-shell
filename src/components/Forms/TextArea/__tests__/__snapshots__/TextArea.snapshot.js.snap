// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea renders TextArea 1`] = `
<div
  className="froms__col"
>
  <p>
    test
  </p>
  <textarea
    className="win-textarea"
    disabled={false}
    name="test"
    onChange={[Function]}
    placeholder="test"
    required={true}
    rows={8}
    style={
      Object {
        "height": "40px",
      }
    }
    value="test"
  />
</div>
`;
